src/transport/labels.rs

```diff
@@ -0,0 +1,114 @@
+use super::tls;
+use crate::metrics::FmtLabels;
+use crate::Conditional;
+use std::fmt;
+
+/// Describes a class of transport.
+///
+/// A `Metrics` type exists for each unique `Key`.
+///
+/// Implements `FmtLabels`.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+pub struct Key {
+    direction: Direction,
+    peer: Peer,
+    tls_status: TlsStatus,
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+struct Direction(&'static str);
```

Comment:
Was `Direction` an `Enum` before? If not, maybe a good time to consider doing so? I only seeing this being `Inbound` or `Outbound` and it could be helpful to have that in the type.

It would also make it more consistent with `Peer` being an `Enum` already.

Reply:
There's another direction enum... all of the labels need to get cleaned up, but let's do that later.